bokeh/models/tools.py

```diff
@@ -778,6 +779,47 @@ class BoxSelectTool(Drag, SelectTool):
     line_dash=[4, 4]
 )
 
+
+class MeasureTool(Drag, Inspection):
+    names = List(String, help="""
+    A list of names to query for. If set, only renderers that have a matching
+    value for their ``name`` attribute will be used.
+    """)
+
+    renderers = Either(Auto, List(Instance(Renderer)), default="auto", help="""
+    An explicit list of renderers to hit test against. If unset,
+    defaults to all renderers on a plot.
+    """)
```

Comment:
IIRC we decided to simply use mouse interactions to define the positions to measure, and not try to do any kind of "snapping to" glyphs (since this could mean different things fore different glyphs, doesn't even make sense at all for some). In which case these `name` and `renderers` properties are not needed at all

Reply:
Yes, I agree, renderers are not of any use for us in this tool. At first, I kept them as it is to use the existing tooltip machinery in a similar way, but that's wrong. I have a more clear picture of the things now. I will edit it.